samples/bluetooth/fast_pair/locator_tag/src/ui/platform_dk.c

```diff
@@ -223,6 +259,44 @@ static void provisioning_led_work_handle(struct k_work *item)
 	dk_set_led(APP_PROVISIONING_LED, provisioning_led_on);
 }
 
+static void ring_led_work_handle(struct k_work *item)
+{
+
+	if (k_work_delayable_is_pending(&motion_led_work)) {
+		/* Do nothing - signalizing motion has higher priority.
+		 * Proper ringing state will be signalized after signalizing motion finishes.
+		 */
+		return;
+	}
+
+	ring_and_motion_led_on = atomic_test_bit(ui_state_status, APP_UI_STATE_RINGING);
+	dk_set_led(APP_RING_AND_MOTION_LED, ring_and_motion_led_on);
+}
+
+static void motion_led_work_handle(struct k_work *item)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(item);
+	static size_t toggle_count;
+	static const size_t toggle_count_max = 4;
+
+	if (motion_detected_led_signal_start) {
+		toggle_count = 0;
+		motion_detected_led_signal_start = false;
+	}
+
+	ring_and_motion_led_on = !ring_and_motion_led_on;
+	toggle_count++;
+
+	if (toggle_count < toggle_count_max) {
+		k_work_reschedule_for_queue(&led_workq, dwork, K_MSEC(100));
+	} else {
+		/* Signal correct ringing state after signalizing motion finishes. */
+		k_work_reschedule_for_queue(&led_workq, &ring_led_work, K_MSEC(100));
+	}
+
+	dk_set_led(APP_RING_AND_MOTION_LED, ring_and_motion_led_on);
```

Comment:
It would be great to control the same LED from just one function. Could we have everything handled in the ring_and_motion_led_work_handle function that merges both the motion_led_work_handle and the ring_led_work_handle functions?

Reply:
There's a problem with rescheduling the work if the ringing state changes while the motion is being signalized. By rescheduling we overwrite the toggle interval. I can make the motion work not change the led directly and submit work to do it with K_NO_WAIT timeout, but I don't think it's better than what we have now. It also introduces some race between two works.